p2p/server.go

```diff
@@ -105,107 +101,171 @@ type Server struct {
 
 	// Hooks for testing. These are useful because we can inhibit
 	// the whole protocol stack.
-	setupFunc
-	newPeerHook
+	newTransport func(net.Conn) transport
+	newPeerHook  func(*Peer)
+
+	lock    sync.Mutex // protects running
+	running bool
 
+	ntab         *discover.Table
+	listener     net.Listener
 	ourHandshake *protoHandshake
 
-	lock         sync.RWMutex // protects running, peers and the trust fields
-	running      bool
-	peers        map[discover.NodeID]*Peer
-	staticNodes  map[discover.NodeID]*discover.Node // Map of currently maintained static remote nodes
-	staticDial   chan *discover.Node                // Dial request channel reserved for the static nodes
-	staticCycle  time.Duration                      // Overrides staticPeerCheckInterval, used for testing
-	trustedNodes map[discover.NodeID]bool           // Set of currently trusted remote nodes
+	// These are for Peers, PeerCount (and nothing else).
+	peerop     chan peerOpFunc
+	peeropResp chan struct{}
+
+	quit          chan struct{}
+	addstatic     chan *discover.Node
+	posthandshake chan *conn
+	addpeer       chan *conn
+	delpeer       chan *Peer
+	loopWG        sync.WaitGroup // loop, listenLoop
+}
 
-	ntab     *discover.Table
-	listener net.Listener
+type peerOpFunc func(map[discover.NodeID]*Peer)
```

Comment:
The name's a bit inconsistent: `peerOpFunc` vs `peerop` and `peeropResp`. I'd go with a single form, doesn't really matter which.